Testing/Readme.md

```diff
@@ -0,0 +1,315 @@
+# ScubaGear Functional Testing Automation <!-- omit in toc --> #
+
+This document outlines the ScubaGear software test automation and its usage. The document also contains instructions for adding new functional tests to existing automation suite.
+
+## Table of Contents <!-- omit in toc --> ##
+
+- [ScubaGear Functional Testing Automation ](#scubagear-functional-testing-automation-)
+  - [Table of Contents ](#table-of-contents-)
+  - [Functional Testing Prerequisites](#functional-testing-prerequisites)
+    - [Windows System with Chrome](#windows-system-with-chrome)
+    - [Pester](#pester)
+    - [Selenium](#selenium)
+    - [Service Principal Account](#service-principal-account)
+  - [Functional Testing Structure](#functional-testing-structure)
+    - [Functional test orchestrator](#functional-test-orchestrator)
+    - [Product test plans](#product-test-plans)
+  - [Functional Testing Usage](#functional-testing-usage)
+    - [Test Usage Example](#test-usage-example)
+  - [Adding New Functional Tests](#adding-new-functional-tests)
+    - [Adding new functional test - Example #1](#adding-new-functional-test---example-1)
+    - [Adding new functional test - Example #2](#adding-new-functional-test---example-2)
+  - [Nightly Functional Tests](#nightly-functional-tests)
+  - [Troubleshooting](#troubleshooting)
+    - [Chrome browser version issue](#chrome-browser-version-issue)
+    - [Service principal authentication issue](#service-principal-authentication-issue)
+
+## Functional Testing Prerequisites ##
+
+Running ScubaGear functional test automation require Windows compute or VM end system. This development system should be setup with Pester, Selenium, Chrome and PowerShell 5.1. The repository provides a few utility scripts to install and update these prerequisite components.
+
+### Windows System with Chrome ###
+
+Functional testing needs to be run on a Windows system with Chrome browser installed. Automation suite uses Selenium tool which requires Chrome driver for its web component testing.
+
+### Pester ###
+
+"Pester is a testing and mocking framework for PowerShell." ([Pester Quick Start](https://pester.dev/docs/quick-start))
+
+On your Windows development system, install an updated Pester module by running following command as an administrator:
+
+```
+Install-Module -Name Pester -Force -SkipPublisherCheck
```

Comment:
When I ran this command, I got the following warning:

`WARNING: The version '5.5.0' of module 'Pester' is currently in use. Retry the operation after closing the
applications.`

Please add this warning to the instructions and include instructions for how to respond to it.

Reply:
Resolved with an additional note in the troubleshooting section.